### PR TITLE
Update fides_disable_save_api option in FidesJS SDK to disable both privacy-preferences & notice-served APIs

### PR DESCRIPTION
**Pull request by NevilleS**
Closes PROD-2017

### Description Of Changes

This is a small change to make the `fides_disable_save_api` option more consistent; in usage, we expected this to disable all save APIs (both notices-served & privacy-preferences), but as of today only the privacy-preferences API is disabled, which was surprising.

As with a lot of PRs, the majority of the changes here are updating the tests 😄 

### Code Changes

* [X] Update FidesJS to disable calling `notices-served` if `fides_disable_save_api` option is `true`
* [X] Update Cypress tests

### Steps to Confirm

* [X] Run the Cookie House demo locally with `?fides_disable_save_api=true` as a query param
* [X] Check the network tab and observe that `notices-served` and `privacy-preferences` are not called

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/privacy-center/cypress/e2e/consent-banner.cy.ts [hunk 1]
```diff
@@ -2150,40 +2151,42 @@ describe("Consent overlay", () => {
     });
   });
 
-  describe("consent reporting", () => {
+  describe("consent reporting APIs (notices-served, privacy-preferences)", () => {
     const historyId1 = "pri_mock_history_id_1";
     const historyId2 = "pri_mock_history_id_2";
+    const buildMockNotices = (): PrivacyNotice[] => [
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/consent-banner.cy.ts:2157]: DRY'd this up a bit to make these specs easier to read and reuse.

File: clients/fides-js/src/lib/hooks.ts [hunk 2]
```diff
@@ -97,15 +97,24 @@ export const useConsentServed = ({
 
   const handleUIEvent = useCallback(
     async (event: FidesEvent) => {
-      // The only time a notices served API call isn't triggered is when
-      // the BANNER is shown or preview mode is enabled. Calls can be triggered for
-      // TCF_BANNER, TCF_OVERLAY, and OVERLAY
+      // Disable the notices-served API if the fides_disable_save_api option is set
+      if (options.fidesDisableSaveApi) {
+        return;
+      }
```
Comment (NevilleS) [inline clients/fides-js/src/lib/hooks.ts:103]: This is all that was actually needed 👍

**Review by NevilleS — COMMENTED** [review]
Couple notes!

File: clients/privacy-center/cypress/e2e/consent-banner.cy.ts [hunk 3]
```diff
@@ -2338,6 +2290,50 @@ describe("Consent overlay", () => {
         });
       });
     });
+
+    it("when fides_disable_save_api option is set, disables notices-served & privacy-preferences APIs", () => {
```
Comment (eastandwestwind) [inline clients/privacy-center/cypress/e2e/consent-banner.cy.ts:2294]: 🚀

**Review by eastandwestwind — APPROVED** [review]
Looks great!

**Pull request merged**